# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1536078163 -7200
#      Tue Sep 04 18:22:43 2018 +0200
# Node ID f6537847b043a453ac2431608f2c7783b59f51d8
# Parent  507cee1b684ebcc4d9333419ed1850ccd1cf0c5c
ginput.m: temporarily disable mouse tools in Qt toolkit (bug #51452)

* ginput.m: set __mouse_mode__ to "none" during ginput execution and restore
  its previous value afterwards.

* Figure.cc (Figure::update): update cursor shape when __mouse_mode__ has
  changed.

diff --git a/libgui/graphics/Figure.cc b/libgui/graphics/Figure.cc
--- a/libgui/graphics/Figure.cc
+++ b/libgui/graphics/Figure.cc
@@ -536,16 +536,20 @@ namespace QtHandles
 
             win->setWindowModality (Qt::ApplicationModal);
             win->setVisible (is_visible);
           }
         else
           win->setWindowModality (Qt::NonModal);
 
         break;
+        
+      case figure::properties::ID___MOUSE_MODE__:
+        m_container->canvas (m_handle)->setCursor (mouseMode ());
+        break;
 
       default:
         break;
       }
 
     m_blockUpdates = false;
   }
 
diff --git a/scripts/plot/util/ginput.m b/scripts/plot/util/ginput.m
--- a/scripts/plot/util/ginput.m
+++ b/scripts/plot/util/ginput.m
@@ -66,23 +66,25 @@ function varargout = ginput (n = -1)
   endif
 
   x = y = button = [];
   ginput_accumulator (0, 0, 0, 0);  # initialize accumulator
 
   orig_windowbuttondownfcn = get (fig, "windowbuttondownfcn");
   orig_keypressfcn = get (fig, "keypressfcn");
   orig_closerequestfcn = get (fig, "closerequestfcn");
+  orig_mousemode = get (fig, "__mouse_mode__");
 
   unwind_protect
 
     set (fig, "windowbuttondownfcn", @ginput_windowbuttondownfcn);
     set (fig, "keypressfcn", @ginput_keypressfcn);
     set (fig, "closerequestfcn", {@ginput_closerequestfcn,
                                   orig_closerequestfcn});
+    set (fig, "__mouse_mode__", "none");
 
     do
       if (strcmp (toolkit, "fltk"))
         __fltk_check__ ();
       endif
 
       ## Release CPU.
       pause (0.01);
@@ -98,16 +100,17 @@ function varargout = ginput (n = -1)
     endif
 
   unwind_protect_cleanup
     if (isfigure (fig))
       ## Only execute if window still exists
       set (fig, "windowbuttondownfcn", orig_windowbuttondownfcn);
       set (fig, "keypressfcn", orig_keypressfcn);
       set (fig, "closerequestfcn", orig_closerequestfcn);
+      set (fig, "__mouse_mode__", orig_mousemode);
     endif
   end_unwind_protect
 
   varargout = {x, y, button};
 
 endfunction
 
 function [rx, ry, rn, rbutton] = ginput_accumulator (mode, xn, yn, btn)
